climate_index_collection/reductions.py

```diff
@@ -410,6 +415,76 @@ def area_mean_weighted(
     return averaged
 
 
+def area_mean_weighted_polygon_selection(
+    dobj,
+    polygon_lon_lat=None,
+    lon_name="lon",
+    lat_name="lat",
+    lat_extent_name=None,
+    lon_extent_name=None,
+):
+    """Area average over polygon-selected region.
+
+    Parameters
+    ----------
+    dobj: xarray.DataArray
+        Data for which the area average will be computed.
+    polygon_lon_lat: shapely Polygon
+        (Multi)Polygon containing the region over which we want to average.
+        If "None", all data will be used.
```

Comment:
Not sure if this behaviour is what will happen.
None isn't handled by the polygon2mask function as far as I can see...
... So it might not return a mask of only `True` values.
Will need to take care of this, if we want this behaviour (which I think is a good idea).